torchacc/dist/fsdp.py

```diff
@@ -202,3 +219,178 @@ def clip_grad_norm_(self, max_grad_norm):
 
     def forward(self, *args, **kwargs):
         return self.model(*args, **kwargs)
+
+    def optim_state_dict(
```

Comment:
Can we implement this like https://pytorch.org/docs/stable/fsdp.html#torch.distributed.fsdp.FullyShardedDataParallel.full_optim_state_dict ? since we don't need to handle the save of sharded optimized state. Use can directly save the sharded  optim state

Reply:
these two funcs should also implement in distributed_parallel.py like https://github.com/AlibabaPAI/torchacc/blob/main/torchacc/dist/distributed_parallel.py#L68

Reply:
> Can we implement this like https://pytorch.org/docs/stable/fsdp.html#torch.distributed.fsdp.FullyShardedDataParallel.full_optim_state_dict ? since we don't need to handle the save of sharded optimized state. Use can directly save the sharded optim state

sure, we can rename the function as full_optim_state_dict and remove the state_dict_type option.